### PR TITLE
Added juniper_junos_show_lldp_neighbors

### PR DESCRIPTION
**Pull request by qduk**
##### ISSUE TYPE

 - New Template Pull Request

##### COMPONENT

Juniper JunOS - show lldp neighbors

##### SUMMARY

 - Adds show lldp neighbors template for JunOS

### CODE REVIEW

File: templates/juniper_junos_show_lldp_neighbors.textfsm [hunk 1]
```diff
@@ -0,0 +1,15 @@
+Value Required LOCAL_INTERFACE (\S+)
+Value Required PARENT_INTERFACE (\S+)
+Value Required CHASSIS_ID (\S+)
+Value Required PORT_INFO (\S*)
+Value Required SYSTEM_NAME (\S+)
+
+Start
```
Comment (FragmentedPacket) [inline templates/juniper_junos_show_lldp_neighbors.textfsm:7]: I would keep everything under `Start` or act off the complete header row to make sure we're going to capture the correct data.

e.g.
```
Start
  ^Local\s+Interface\s+Parent\s+Interface\s+Chassis\s+Id\s+Port\s+info\s+System\s+Name
  ^blah\s+blah\s+blah

or

Start
  ^Local\s+Interface\s+Parent\s+Interface\s+Chassis\s+Id\s+Port\s+info\s+System\s+Name -> LLDP
```

I hope that makes sense.

File: development_scripts.py [hunk 2]
```diff
@@ -289,7 +289,7 @@ def transform_file(filepath):
     Example:
         >>> filepath = "tests/cisco_ios/show_version/cisco_ios_show_version.yml"
         >>> transform_parsed(filepath)
-        >>> 
+        >>>
```
Comment (FragmentedPacket) [inline development_scripts.py:292]: Do you mind removing the two files that aren't related to the templates?

`git checkout master development_scripts.py`

File: templates/juniper_junos_show_lldp_neighbors.textfsm (hunk 1 above)
Comment (FragmentedPacket) [reply templates/juniper_junos_show_lldp_neighbors.textfsm:7]: If you stick with the two separate states, each state requires the following:
```
  ^\s*$$
  ^. -> Error
```

File: templates/juniper_junos_show_lldp_neighbors.textfsm [hunk 3]
```diff
@@ -0,0 +1,15 @@
+Value Required LOCAL_INTERFACE (\S+)
+Value Required PARENT_INTERFACE (\S+)
+Value Required CHASSIS_ID (\S+)
+Value Required PORT_INFO (\S*)
```
Comment (FragmentedPacket) [inline templates/juniper_junos_show_lldp_neighbors.textfsm:4]:
```suggestion
Value Required PORT_INFO (\S+)
```

File: templates/juniper_junos_show_lldp_neighbors.textfsm [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+Value Required SYSTEM_NAME (\S+)
+
+Start
+  ^Local\s+Interface -> LLDP
+
+LLDP
+  ^${LOCAL_INTERFACE}\s*${PARENT_INTERFACE}\s+${CHASSIS_ID}\s+${PORT_INFO}\s+${SYSTEM_NAME} -> Record
+  ^${LOCAL_INTERFACE}\s*
+  ^${PARENT_INTERFACE}\s+${CHASSIS_ID}\s+${PORT_INFO}\s+${SYSTEM_NAME} -> Record
```
Comment (FragmentedPacket) [inline templates/juniper_junos_show_lldp_neighbors.textfsm:13]: Do you have an example of the local interface on a line and then the rest of it on a separate line? If so, do you mind adding it to the `.raw` file.

File: tests/test_testcases_exists.py [hunk 5]
```diff
@@ -11,8 +11,7 @@
 
 
 def extract_index_data():
-    """Used to parametrize and report each test case with the necessary data.
-    """
+    """Used to parametrize and report each test case with the necessary data."""
```
Comment (FragmentedPacket) [inline tests/test_testcases_exists.py:14]: `git checkout master tests/test_testcases_exists.py`

**Review by FragmentedPacket — CHANGES_REQUESTED** [review]

File: templates/juniper_junos_show_lldp_neighbors.textfsm (hunk 1 above)
Comment (qduk) [reply templates/juniper_junos_show_lldp_neighbors.textfsm:7]: Added the complete header and appended 
```
  ^\s*$$
  ^. -> Error
```
to the LLDP state.

File: templates/juniper_junos_show_lldp_neighbors.textfsm (hunk 3 above)
Comment (qduk) [reply templates/juniper_junos_show_lldp_neighbors.textfsm:4]: Updated regex expression for PORT_INFO

File: templates/juniper_junos_show_lldp_neighbors.textfsm (hunk 4 above)
Comment (qduk) [reply templates/juniper_junos_show_lldp_neighbors.textfsm:13]: Per our discussion, I did not have an example of local interface on one line and the rest on another. So I removed lines 12 and 13.

File: tests/test_testcases_exists.py (hunk 5 above)
Comment (qduk) [reply tests/test_testcases_exists.py:14]: Command ran.

File: development_scripts.py (hunk 2 above)
Comment (qduk) [reply development_scripts.py:292]: Command ran.

File: templates/juniper_junos_show_lldp_neighbors.textfsm (hunk 1 above)
Comment (qduk) [reply templates/juniper_junos_show_lldp_neighbors.textfsm:7]: Condensed everything to one state.

**Comment by FragmentedPacket** [comment]
Did you push up to your forked branch? I only see the single commit still.

**Review by FragmentedPacket — APPROVED** [review]

**Pull request merged**